Add CloseWatcher reference pages (#33910)
mdn/content: files/en-us/web/api/closewatcher/destroy/index.md

---
title: "CloseWatcher: destroy() method"
short-title: destroy()
slug: Web/API/CloseWatcher/destroy
page-type: web-api-instance-method
status:
  - experimental
browser-compat: api.CloseWatcher.destroy
---

{{APIRef("HTML DOM")}} {{SeeCompatTable}}

> The **`destroy()`** method of the {{domxref("CloseWatcher")}} interface deactivates the close watcher so that it will no longer receive `close` events.

Reviewer: ```suggestion
The **`destroy()`** method of the {{domxref("CloseWatcher")}} interface deactivates the close watcher so that it will no longer receive `cancel` or `close` events.
```

?

Reviewer: Again the spec gives some helpful guidance and context:

> Deactivates watcher, so that it will no longer receive [close](https://html.spec.whatwg.org/multipage/indices.html#event-close) events and so that new independent [CloseWatcher](https://html.spec.whatwg.org/multipage/interaction.html#closewatcher) instances can be constructed.
>
> This is intended to be called if the relevant UI element is torn down in some other way than being closed.

## Syntax

```js-nolint
destroy()
```

### Parameters

None.

### Return value

None ({{jsxref("undefined")}}).

## Examples

### Using the `destroy()` method

> Use the `destory()` method to dispose the watcher instance

Reviewer: ```suggestion
Use the `destroy()` method to dispose of the watcher instance
```

```js
// Listen for close requests
watcher.addEventListener("close", () => {
  // Close your UI component
  sidebar.hide();

>   // Dispose the watcher instance

Reviewer: ```suggestion
  // Dispose of the watcher instance
```
